alicloud/resource_alicloud_logtail_to_machine_group_test.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+	"strings"
+	"testing"
+)
+
+func TestAccAlicloudLogtailToMachineGroup_basic(t *testing.T) {
+	var project sls.LogProject
+	var group sls.MachineGroup
+	var store sls.LogStore
+	var config sls.LogConfig
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAlicloudLogtailToMachineGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudLogtailToMachineGroup_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudLogProjectExists("alicloud_log_project.test", &project),
+					testAccCheckAlicloudLogStoreExists("alicloud_log_store.test", &store),
+					testAccCheckAlicloudLogTailConfigExists("alicloud_logtail_config.test", &config),
+					testAccCheckAlicloudLogMachineGroupExists("alicloud_log_machine_group.test", &group),
+					testAccCheckAlicloudLogtailToMachineGroupExists("alicloud_logtail_to_machine_group.test", config.Name),
+					resource.TestCheckResourceAttr("alicloud_logtail_to_machine_group.test", "project", "test-tf2"),
+					resource.TestCheckResourceAttr("alicloud_logtail_to_machine_group.test", "logtail_config_name", "evan-terraform-update"),
+					resource.TestCheckResourceAttr("alicloud_logtail_to_machine_group.test", "machine_group_name", "evan-machine-group"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAlicloudLogtailToMachineGroupExists(name, group_name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return WrapError(fmt.Errorf("Not found: %s", name))
+		}
+
+		if rs.Primary.ID == "" {
+			return WrapError(fmt.Errorf("No Log machine group ID is set"))
+		}
+
+		split := strings.Split(rs.Primary.ID, COLON_SEPARATED)
+
+		client := testAccProvider.Meta().(*connectivity.AliyunClient)
+		logService := LogService{client}
+		groupNames, err := logService.DescribeLogtailToMachineGroup(split[0], split[1])
+		if err != nil {
+			return WrapError(err)
+		}
+		for _, name := range groupNames {
+			if name == split[2] {
+				group_name = name
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckAlicloudLogtailToMachineGroupDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*connectivity.AliyunClient)
+	logService := LogService{client}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "" {
+			continue
+		}
+
+		split := strings.Split(rs.Primary.ID, COLON_SEPARATED)
+
+		if _, err := logService.DescribeLogMachineGroup(split[0], split[1]); err != nil {
+			if NotFoundError(err) {
+				continue
+			}
+			return WrapError(err)
+		}
+		return WrapError(fmt.Errorf("Logtail to machine group %s still exists.", rs.Primary.ID))
+	}
+	return nil
+}
+
+const testAccCheckAlicloudLogtailToMachineGroup_basic = `
+resource "alicloud_log_project" "test"{
+	name = "test-tf2"
```

Comment:
Please set a name with prefix 'tf-testAcc'. And also use a rand to make the project is different.